diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2d5f09..6de3158 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,9 @@ add_library(galaxy-fds-sdk-cpp ${DIR_SRCS})
 target_include_directories(galaxy-fds-sdk-cpp PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 include_directories(${POCO_INCLUDE})
+if (APPLE)
+    include_directories(${MACOS_OPENSSL_INCLUDE_DIR})
+endif()
 link_directories(${POCO_LIB})
 
 option(WITH_TEST "enable building test" OFF)
