diff --git a/hadoop-hdfs-project/hadoop-hdfs-native-client/src/CMakeLists.txt b/hadoop-hdfs-project/hadoop-hdfs-native-client/src/CMakeLists.txt
index 24ec297aa27b..e77c38435bba 100644
--- a/hadoop-hdfs-project/hadoop-hdfs-native-client/src/CMakeLists.txt
+++ b/hadoop-hdfs-project/hadoop-hdfs-native-client/src/CMakeLists.txt
@@ -152,24 +152,13 @@ add_subdirectory(main/native/libhdfs-examples)

 # Temporary fix to disable Libhdfs++ build on older systems that do not support thread_local
 include(CheckCXXSourceCompiles)
-unset (THREAD_LOCAL_SUPPORTED CACHE)
 set (CMAKE_REQUIRED_DEFINITIONS "-std=c++11")
 set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
-check_cxx_source_compiles(
-    "#include <thread>
-    int main(void) {
-      thread_local int s;
-      return 0;
-    }"
-    THREAD_LOCAL_SUPPORTED)
-if (THREAD_LOCAL_SUPPORTED)
+if (REQUIRE_LIBHDFSPP)
     add_subdirectory(main/native/libhdfspp)
 else()
-    message(WARNING
-    "WARNING: Libhdfs++ library was not built because the required feature thread_local storage \
-    is not supported by your compiler. Known compilers that support this feature: GCC 4.8+, Visual Studio 2015+, \
-    Clang (community version 3.3+), Clang (version for Xcode 8+ and iOS 9+).")
-endif (THREAD_LOCAL_SUPPORTED)
+    message(WARNING "WARNING: Libhdfs++ library was not built because the REQUIRE_LIBHDFSPP is not enabled.")
+endif (REQUIRE_LIBHDFSPP)

 if(REQUIRE_LIBWEBHDFS)
     add_subdirectory(contrib/libwebhdfs)